src/doc/rustc/src/platform-support/fuchsia.md

```diff
@@ -31,20 +31,132 @@ from the [source tree] or using the Fuchsia SDK.
 
 [source tree]: https://fuchsia.dev/fuchsia-src/get-started/learn/build
 
-Fuchsia targets support std and follow the `sysv64` calling convention on
+Fuchsia targets support `std` and follow the `sysv64` calling convention on
 x86_64. Fuchsia binaries use the ELF file format.
 
-## Building the target
+You'll also need a recent copy of the [Fuchsia SDK], which provides the tools, libraries, and  binaries required to build and link programs for Fuchsia.
+
+[Fuchsia SDK]: https://chrome-infra-packages.appspot.com/p/fuchsia/sdk/core
+
+## Walkthrough structure
+
+This walkthrough will cover:
+
+1. Compiling a Rust binary targeting Fuchsia.
+1. Building a Fuchsia package.
+1. Publishing and running a Fuchsia package to a Fuchsia emulator.
+
+For the purposes of this walkthrough, we will only target `x86_64-fuchsia`.
+
+## Compiling a Rust binary targeting Fuchsia
+
+Today, there are two main ways to build a Rust binary targetting Fuchsia:
+
+1. Allowing [rustup] to handle the installation of Fuchsia targets for you,
+   and using the [cargo] tool to build the Rust binary.
+1. Using the Fuchsia SDK and related `sysroot` to build and install a `rustc`
+   that can target Fuchsia.
+
+[rustup]: (https://rustup.rs/)
+[cargo]: (https://doc.rust-lang.org/cargo/)
+
+### Compiling a Rust binary targeting Fuchsia (Rustup/Cargo)
+
+The easiest way to build a Rust binary targeting Fuchsia is by allowing [rustup]
+to handle the installation of Fuchsia targets for you. This can be done by issuing
+the following commands:
+
+```sh
+rustup target add x86_64-fuchsia
+rustup target add aarch_64-fuchsia
+```
+
+After installing our Fuchsia targets, we can now compile a Rust binary that targets
+Fuchsia.
+
+To create our Rust project, we can issue a standard `cargo` command as follows:
+
+**From base working directory**
+```sh
+cargo new --name hello_fuchsia --bin hello_fuchsia_rust_crate
+```
+
+We can edit our `hello_fuchsia_rust_crate/src/main.rs` to include a test as follows:
+
+**`hello_fuchsia_rust_crate/src/main.rs`**
+```rust
+fn main() {
+    println!("Hello Fuchsia!");
+}
+
+#[test]
+fn it_works() {
+    assert_eq!(2 + 2, 4);
+}
+```
+
+In addition to the standard workspace created, we will want to create a
+`hello_fuchsia_rust_crate/build.rs` file to link necessary libraries during compilation:
+
+**`hello_fuchsia_rust_crate/build.rs`**
+```rust
+fn main() {
+  println!("cargo:rustc-link-search=native=<SDK_PATH>/arch/x64/sysroot/lib");
+  println!("cargo:rustc-link-search=native=<SDK_PATH>/arch/x64/lib");
+}
+```
+
+*Note: Make sure to fill out `<SDK_PATH>` with the path to the downloaded SDK.*
+
+In total, our new project will look like:
+
+**Current directory structure**
+```txt
+hello_fuchsia_rust_crate/
+┣━ src/
+┃  ┗━ main.rs
+┣━ cargo.toml
+┗━ build.rs
+```
+
+Finally, we can build our rust binary as:
+
+```sh
+cargo build --release --target x86_64-fuchsia --manifest-path hello_fuchsia_rust_crate/Cargo.toml
```

Comment:
For the walkthrough, I've built all commands to be run from the same base directory. Since `cargo new` creates the hello_fuchsia_rust_crate/ directory, it can't find the Cargo.toml unless specified

Agreed on debug change!